aten/src/ATen/native/cuda/jit_utils.cpp

```diff
@@ -280,10 +280,30 @@ const std::string dynamic_cast_support_literal = R"ESCAPE(
     }
   };
 
+  template <typename T>
+  struct LoadImpl {
+    __device__ static T apply(const void *src) {
+      return *reinterpret_cast<const T*>(src);
+    }
+  };
+
+  template <>
+  struct LoadImpl<bool> {
```

Comment:
@ngimel and I reviewing together:

Could we see before/after conversion times? Like bool->long, bool->float, and long->float (to see if non-bool conversions are affected)?

Reply:
Could we also see the perf delta on some TensorIterator kernels, like torch.add with bool x long, bool x float, and long x float?